File: 1679-max-number-of-k-sum-pairs/1679-max-number-of-k-sum-pairs.java
class Solution {
    public int maxOperations(int [] nums, int k) {
		int j = nums.length - 1, i = 0;
		int count = 0;
		Arrays.sort(nums);
	   while (i < j)
	   {
		   if (nums[i] + nums[j] == k)
		   {
			   i++; j--; count++;
		   } else if (nums[i] + nums[j] > k)
			   j--;
		   else
			   i++;
	   }
	   return count;
	}
}

File: 1679-max-number-of-k-sum-pairs/1679-max-number-of-k-sum-pairs.py
class Solution:
    def maxOperations(self, nums: List[int], k: int) -> int:
        l, r = 0, len(nums) - 1
        count = 0
        nums.sort()
        while l < r:
            if nums[l] + nums[r] == k:
                count += 1
                l += 1
                r -= 1
            # Adding the largest number to the smallest number yield a higher number than the target,
            # Then there is no other number added with the largest number that would give us the target,
            # Any other number added with the largest would result in an even higher number

            # The same is true for the smallest,
            # If the smallest added with the greatest yield a smaller number than the target
            # There is no other number that when added resulted in a larger number,
            # Thus when this situation came, we remove them from the list
            elif nums[l] + nums[r] > k:
                r -= 1
            else:
                l += 1

        return count